src/routes/data.ts

```diff
@@ -0,0 +1,187 @@
+import { badGateway } from "@hapi/boom";
+import joi from "joi";
+import { PutObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { DataFormat, MIME_TYPE, StorageService } from "../constants.js";
+import { UnixFS } from "@web3-storage/upload-client";
+import type { Client as W3UpClient } from "@web3-storage/w3up-client";
+import type {
+    Lifecycle,
+    ReqRefDefaults,
+    ResponseToolkit,
+    ServerRoute,
+    Request,
+} from "@hapi/hapi";
+
+interface GetDataRoutesParams {
+    w3UpClient: W3UpClient;
+    s3Client: S3Client;
+    s3Bucket: string;
+}
+
+type DataFormatSerializer = (deserialized: any) => Promise<string> | string;
+
+type StorageServiceHandler = (
+    storageFormat: DataFormat,
+    request: Request<ReqRefDefaults>,
+    h: ResponseToolkit,
+) => Promise<Lifecycle.ReturnValue>;
+
+export const getDataRoutes = async ({
+    w3UpClient,
+    s3Client,
+    s3Bucket,
+}: GetDataRoutesParams): Promise<ServerRoute[]> => {
+    const SUPPORTED_FORMAT_SERIALIZERS: Record<
+        DataFormat,
+        DataFormatSerializer
+    > = {
+        [DataFormat.Json]: JSON.stringify,
+    };
+
+    const SUPPORTED_STORAGE_SERVICE_HANDLERS: Record<
+        StorageService,
+        StorageServiceHandler
+    > = {
+        [StorageService.Ipfs]: async (format, request, h) => {
+            const { data } = request.payload as { data: object };
+            const dataString = await SUPPORTED_FORMAT_SERIALIZERS[format](data);
```

Comment:
We can omit the `await` here, since `JSON.stringify` doesn't return a promise.

Reply:
The problem is that `SUPPORTED_FORMAT_SERIALIZERS` is defined as `Record<DataFormat, DataFormatSerializer>`, with `DataFormatSerializer` being defined as `type DataFormatSerializer = (deserialized: any) => Promise<string> | string;`. As you can see the return value here can be a promise, so we need the `await`. In case the return value is a simple string it will just do nothing.